tests/phpunit/tests/admin/includesPost.php

```diff
@@ -686,6 +686,34 @@ public function test_get_sample_permalink_should_respect_hierarchy_of_draft_page
 		$this->assertSame( 'child-page', $actual[1] );
 	}
 
+	/**
+	 * Test if get_sample_permalink preserves the WP_Post properties.
+	 *
+	 * @ticket 54736
+	 */
+	public function test_get_sample_permalink_should_preserve_the_original_post_properties() {
+		$post = self::factory()->post->create_and_get(
+			array(
+				'post_status' => 'draft',
+			)
+		);
+
+		$post_original = clone $post;
+
+		add_filter(
+			'get_sample_permalink',
+			function( $permalink, $post_id, $title, $name, $post ) use ( $post_original ) {
+				$this->assertEquals( $post_original, $post, 'Modified post object passed to get_sample_permalink filter.' );
+				return $permalink;
+			},
+			10,
+			5
+		);
+
+		get_sample_permalink( $post );
+		$this->assertEquals( $post_original, $post, 'get_sample_permalink() modifies the post object.' );
```

Comment:
This can't be changed for now and is the right assertion, but I just want to say that after going through the test suite and proposing a PR to reduce `assertEquals()` from 500+ to ~30, seeing `assertEquals()` kills me. You've killed me. Murderer.